src/components/feedback/test/Dialog-test.js

```diff
@@ -19,7 +20,14 @@ const createComponent = (Component, props = {}) => {
  */
 const ComponentWithTransition = ({ children, direction, onTransitionEnd }) => {
   // Fake a 50ms transition time
-  setTimeout(() => onTransitionEnd?.(direction), 50);
+  useEffect(() => {
+    const timeout = setTimeout(() => onTransitionEnd?.(direction), 50);
+
+    return () => {
+      clearTimeout(timeout);
+    };
+  }, [direction, onTransitionEnd]);
```

Comment:
This stub `TransitionComponent` was triggering this timeout on every render, which was not relevant for a component that was previously mounted and unmounted every time.

With the new logic to open/close, the component can render multiple times, triggering the setTimeout an incorrect amount of times. Hence, the `useEffect`.